### PR TITLE
Split system tests for API 300

### LINKED ISSUE #109: System Tests Update for Support API 300
### Scenario/Intent
System tests are not suitable for the API 300. You must modify the test strategy to run tests for different schemas.

### Expected Result
System tests running for API200, API300(C7000 and Synergy).

### Actual Result
System tests running only for API200.

### PR DESCRIPTION
**Pull request by aalexmonteiro**
### Description
System tests are suitable for the API 300. Test strategy to run tests for different schemas. Please read the README.md file from the spec/system/README.md file for instructions.

### Issues Resolved
#109 

### Check List
- [ ] New functionality includes testing.
  - [] All tests pass (`$ rake test`).
- [x] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in the examples (please include helpful comments).
- [ ] Changes are documented in the CHANGELOG.

### CODE REVIEW

File: spec/shared_context.rb [hunk 1]
```diff
@@ -121,3 +116,30 @@ def integration_context_debugging
     end
   end
 end
+
+def system_context
```
Comment (tmiotto) [inline spec/shared_context.rb]: The name of this method is misleading... This is not only the system context, it also loads a lot of common vars, secrets, clients...

Please split it in two different methods.

Also, you left some of the *shared client load and mock* dangling in the `before(:each)`. Please bring it here before the method split.

File: spec/shared_context.rb [hunk 2]
```diff
@@ -47,27 +47,22 @@
...
-    end
-
-    $secrets ||= OneviewSDK::Config.load(@secrets_path) # Secrets for URIs, server/enclosure credentials, etc.
-
-    # Create client objects:
-    $config  ||= OneviewSDK::Config.load(@config_path)
+    system_context
     $client_120 ||= OneviewSDK::Client.new($config.merge(api_version: 120))
     $client     ||= OneviewSDK::Client.new($config.merge(api_version: 200))
+
```
Comment (tmiotto) [inline spec/shared_context.rb]: See the comment on the `system_context` method

File: spec/shared_context.rb (hunk 1 above)
Comment (aalexmonteiro) [inline spec/shared_context.rb]: @tmiotto I have a question about something you mentioned:

Also, you left some of the shared client load and mock hanging on before (each:). Please bring it here before splitting the method.

I've removed the clients from before(:each), but is it really necessary to remove the mocks? For outside, the same would not work. Maybe I did not quite understand what you said.

Comment (tmiotto) [inline spec/shared_context.rb]: Do not remove them, just move to another method, or leave it there.
The problem is you start this (semantic) block in the method:
```ruby
  # Create client objects:
  $config ||= OneviewSDK::Config.load(@config_path)
  $config_synergy ||= OneviewSDK::Config.load(@config_path_synergy)
end
```

And then finishes it in the `before(:each)`:
```ruby
    system_context # You started the 'create client objects' inside this method 
    $client_120 ||= OneviewSDK::Client.new($config.merge(api_version: 120))
    $client     ||= OneviewSDK::Client.new($config.merge(api_version: 200))

    allow_any_instance_of(OneviewSDK::Client).to receive(:appliance_api_version).and_call_original
    allow_any_instance_of(OneviewSDK::Client).to receive(:login).and_call_original
```

My point is that you're splitting in the middle of a semantic block that should not be splitted, these lines should be together, they make sense to be together.

So or you create a method `load_client_configuration` and put all of these lines inside the method, or you leave it all on the before(:each).

Right now, it just seems and arbitrary split in the code to make rubocop not complain, that is breaking the logical flow.

**Comment by aalexmonteiro** [comment]
@tmiotto fixed review.

**Pull request merged**